vortex-array/src/array/chunked/compute/boolean.rs

```diff
@@ -14,21 +14,53 @@ impl BinaryBooleanFn<ChunkedArray> for ChunkedEncoding {
     ) -> VortexResult<Option<ArrayData>> {
         let mut idx = 0;
         let mut chunks = Vec::with_capacity(lhs.nchunks());
+        let mut nullability = Nullability::Nullable;
 
         for chunk in lhs.chunks() {
             let sliced = slice(rhs, idx, idx + chunk.len())?;
-            let result = match op {
-                BinaryOperator::And => and(&chunk, &sliced),
-                BinaryOperator::AndKleene => and_kleene(&chunk, &sliced),
-                BinaryOperator::Or => or(&chunk, &sliced),
-                BinaryOperator::OrKleene => or_kleene(&chunk, &sliced),
-            };
-            chunks.push(result?);
+            let result = binary_boolean(&chunk, &sliced, op)?;
+            nullability = result.dtype().nullability();
```

Comment:
I think this is not quite correct - if the any of the chunks is nullable the chunked array should be nullable. We can move this out of the loop and add

```
        let nullable = chunks.iter().any(|c| c.dtype().is_nullable());
```

Reply:
I forgot that part, then we should be able to take the first chunk's dtype

Reply:
No, nullability should be `lhs.dtype().is_nullable() || rhs.dtype().is_nullable()`

Reply:
I don’t think this works this way for all the Boolean logic functions.

Reply:
My understanding is that the data type (dtype) of the resulting chunked array matches that of its children. These children consist of the outcomes from each `binary_boolean` call, indicating that the dtype of the first chunk is relevant. Perhaps we could replicate the dtype of the first chunk using `ChunkedArray::try_new(chunks, first_chunk.dtype().clone())?.into_array()`.

Reply:
I think what @gatesn suggested is the right answer. There might not be any chunks to take nullability from

Reply:
I agree. It's same with arrow_boolean.